sys/net/application_layer/unicoap/state.c

```diff
@@ -0,0 +1,585 @@
+    case UNICOAP_CODE_CLASS_REQUEST: {
+        if (truncated) {
+            SERVER_DEBUG("truncated, not processing, sending Size1\n");
+            UNICOAP_OPTIONS_ALLOC(response_options, 5);
+            unicoap_options_set_size1(&response_options, CONFIG_UNICOAP_BLOCK_SIZE);
+            unicoap_response_init_with_options(message, UNICOAP_STATUS_REQUEST_ENTITY_TOO_LARGE,
+                                               NULL, 0, &response_options);
+            goto server_prewarming_error;
+        }
+
+        const unicoap_resource_t* resource = NULL;
+        const unicoap_listener_t* listener = NULL;
+        if ((res = unicoap_resource_find(packet, &resource, &listener)) != 0) {
+            unicoap_response_init_empty(message, res < 0 ? UNICOAP_STATUS_INTERNAL_SERVER_ERROR :
+                                                           (unicoap_status_t)res);
+            goto server_prewarming_error;
+        }
+        arg->resource = resource;
+        *flags = _messaging_flags_resource(resource->flags);
+        return UNICOAP_PREPROCESSING_SUCCESS_REQUEST;
+
+server_prewarming_error:
+        unicoap_messaging_send(packet, UNICOAP_MESSAGING_FLAGS_DEFAULT);
+        return UNICOAP_PREPROCESSING_ERROR_REQUEST;
+    }
+
+    case UNICOAP_CODE_CLASS_RESPONSE_SUCCESS:
+    case UNICOAP_CODE_CLASS_RESPONSE_CLIENT_FAILURE:
+    case UNICOAP_CODE_CLASS_RESPONSE_SERVER_FAILURE:
+        /* TODO: Client: Process response */
+        return UNICOAP_PREPROCESSING_ERROR_RESPONSE_UNEXPECTED;
+
+    case UNICOAP_CODE_CLASS_SIGNAL:
+        /* TODO: Signaling */
+        return UNICOAP_PREPROCESSING_ERROR_UNSUPPORTED;
+
+    default:
+        MESSAGING_DEBUG("illegal code class %" PRIu8 "\n", unicoap_code_class(message->code));
+        return UNICOAP_PREPROCESSING_ERROR_INVALID_CODE_CLASS;
+    }
+}
+
+int unicoap_exchange_process(unicoap_packet_t* packet, unicoap_exchange_arg_t arg) {
+    switch (unicoap_code_class(packet->message->code)) {
+    case UNICOAP_CODE_CLASS_REQUEST:
+        return unicoap_server_process_request(packet, arg.resource);
+
+    case UNICOAP_CODE_CLASS_RESPONSE_SUCCESS:
+    case UNICOAP_CODE_CLASS_RESPONSE_CLIENT_FAILURE:
+    case UNICOAP_CODE_CLASS_RESPONSE_SERVER_FAILURE:
+        /* TODO: Client: Process response */
+        return -1;
+
+    case UNICOAP_CODE_CLASS_SIGNAL:
+        /* TODO: Signaling */
+        return -1;
+
+    default:
+        UNREACHABLE();
+        return -1;
+    }
+}
+
+void unicoap_exchange_forget_failed_endpoint(const unicoap_endpoint_t* endpoint)
+{
+    (void)endpoint;
+    /* TODO: Client and advanced server features: Elaborate state management */
+    /* TODO: Observe: Remove potential registrations */
+    return;
+}
+
+int unicoap_options_set_observe_generated(unicoap_options_t* options)
+{
+    /* generate initial notification value */
+    return unicoap_options_set_observe(
+        options, (ztimer_now(ZTIMER_MSEC) >> UNICOAP_OBS_TICK_EXPONENT) & 0xFFFFFF);
```

Comment:
```suggestion
        options, (ztimer_now(UNICOAP_CLOCK) >> UNICOAP_OBS_TICK_EXPONENT) & 0xFFFFFF);
```
? Also, why is this part of `state.c`?